src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java

```diff
@@ -353,9 +368,11 @@ static void processPileupAlleles(AssemblyRegion region, List<VariantContext> giv
                 .collect(Collectors.toList()));
 
         Map<Kmer, Integer> kmerReadCounts = getKmerReadCounts(region.getHardClippedPileupReads(), 10);
-        //TODO BG & AH filter out low base and/or mapping quality alleles
 
-        for (final VariantContext givenVC : givenAlleles) {
+        // Remove SNPs that are too close to assembled indels.
+        final List<VariantContext> givenAllelesFiltered = givenAlleles.stream().filter(vc -> vc.isIndel() || assembledIndels.stream().noneMatch(indel -> vc.withinDistanceOf(indel, snpAdjacentToIndelLimit))).collect(Collectors.toList());
```

Comment:
what's the motivation behind this filter?

Reply:
Its one of the DRAGEN heuristics they use to try to cut down on false positives generated from this code. The idea is if you are in the vicinity of Indels its likely some of the SNPs are alignment mirages anyways so they shouldn't be trusted. It's unclear how necessary this will be in our version given that we don't have their faster HMM code.